### PR TITLE
We don't need no dynamics

### LINKED ISSUE #97: Use strongly typed objects instead of dynamic everywhere
This will probably be post v1.0 but just to keep track of this

### PR DESCRIPTION
**Pull request by kblok**
closes #97

### CODE REVIEW

File: lib/PuppeteerSharp/ExecutionContext.cs [hunk 1]
```diff
@@ -173,8 +174,8 @@ internal async Task<JSHandle> EvaluateFunctionHandleAsync(string script, params
             }
         }
 
-        internal JSHandle CreateJSHandle(dynamic remoteObject)
-            => (remoteObject.subtype == "node" && Frame != null)
+        internal JSHandle CreateJSHandle(JToken remoteObject)
```
Comment (Meir017) [inline lib/PuppeteerSharp/ExecutionContext.cs]: do we want to create a class for this?

File: lib/PuppeteerSharp/Messaging/EvaluateHandleResponse.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using Newtonsoft.Json.Linq;
+
+namespace PuppeteerSharp.Messaging
+{
+    internal class EvaluateHandleResponse
+    {
+        public JToken ExceptionDetails { get; set; }
```
Comment (Meir017) [inline lib/PuppeteerSharp/Messaging/EvaluateHandleResponse.cs]: shouldn't this be a class?

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -1865,7 +1866,7 @@ private async Task<object> ExecuteBinding(BindingCalledResponse e)
                 if (taskResult.GetType().IsGenericType)
                 {
                     // the task is already awaited and therefore the call to property Result will not deadlock
-                    result = ((dynamic)taskResult).Result;
+                    result = taskResult.GetType().GetProperty(taskResultPropertyName).GetValue(taskResult);
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: the old way was weird but this also feels a bit odd, is there a clean way to do this?

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -1904,13 +1905,13 @@ private async Task OnAttachedToTarget(MessageEventArgs e)
...
 
-        private void OnLogEntryAdded(LogEntryAddedResponse e)
+        private async Task OnLogEntryAddedAsync(LogEntryAddedResponse e)
         {
             if (e.Entry.Args != null)
             {
                 foreach (var arg in e.Entry?.Args)
                 {
-                    RemoteObjectHelper.ReleaseObject(Client, arg, _logger);
+                    await RemoteObjectHelper.ReleaseObjectAsync(Client, arg, _logger);
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: is this missing `.ConfigureAwait(false)` ?

File: lib/PuppeteerSharp/ExecutionContext.cs (hunk 1 above)
Comment (Meir017) [reply lib/PuppeteerSharp/ExecutionContext.cs]: fair enough

Comment (kblok) [reply lib/PuppeteerSharp/ExecutionContext.cs]: I will implement a class @Meir017 the shape will be a composition of that ;)

File: lib/PuppeteerSharp/Page.cs (hunk 3 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: I open to new ideas 😂

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Created an issue for this https://github.com/kblok/puppeteer-sharp/issues/824

**Comment by kblok** [comment]
Ready to review again @Meir017

**Pull request merged**